test/integration_tests/conftest.py

```diff
@@ -0,0 +1,28 @@
+import shutil
+
+import pytest
+import torch
+
+
+def pytest_addoption(parser):
+    parser.addoption(
+        "--use-tmp-hub-dir",
+        action="store_true",
+        help=(
+            "When provided, tests will use temporary directory as Torch Hub directory. "
+            "Downloaded models will be deleted after each test."
+        ),
+    )
+
+
+@pytest.fixture(scope="class")
+def temp_hub_dir(tmp_path_factory, pytestconfig):
+    if not pytestconfig.getoption("--use-tmp-hub-dir"):
```

Comment:
Follow-up in #1889 aside, I don't understand how this can work at all.

Both `tmp_path_factory` and `pytestconfig` appear _nowhere else_ in the code base (as far as GH search and `git grep` go, at least), and based on the pytest [docs](https://docs.pytest.org/en/7.2.x/example/simple.html#pass-different-values-to-a-test-function-depending-on-command-line-options) this should probably be
```suggestion
@pytest.fixture(scope="class")
def temp_hub_dir(tmp_path_factory, request):
    if not request.config.getoption("--use-tmp-hub-dir"):
```
but that still leaves me mystified as to where `tmp_path_factory` is coming from.

Reply:
> but that still leaves me mystified as to where `tmp_path_factory` is coming from.

Ah, just found that it's a fixture that comes with pytest by [default](https://docs.pytest.org/en/7.2.x/how-to/tmp_path.html#tmp-path-factory-example).

Reply:
OK, and so's `pytestconfig`. In any case, my suspicion now is that this only works when called from _within_ the `integration_tests` folder, because (I guess) recursive `conftest.py`s are not evaluated when the CLI is called from a folder that's further out.

Reply:
@h-vetinari yup I think you're understanding is correct. Does only being able to run this test from within the `integration_tests` folder pose an issue for adding torchtext to `conda-forge`?

Reply:
> Does only being able to run this test from within the `integration_tests` folder pose an issue for adding torchtext to `conda-forge`?

No, I just had to realise the reason, after that it's simple. ;)